fix(http): fix auth in document handler

affo opened the pull request.

lyondhill commented on authorizer/document.go:
```diff
@@ -0,0 +1,201 @@
+}
+
+func newDocumentOrgPermission(a influxdb.Action, orgID influxdb.ID) (*influxdb.Permission, error) {
+	return influxdb.NewPermission(a, influxdb.DocumentsResourceType, orgID)
```
```suggestion
	return newOrgPermission(a, orgID)
```

affo commented:
@lyondhill 

What I did:
 - removed the possibility to generate side effects with the `DocumentIndex`.
 - make it so that that behavior is now in `kv`.
 - separated authorization layer in `authorizer` package.
 - changed the `authorizer.LabelService` so that it accounts for the special case of documents permissions.
 - fixed tests
